mathesar/tests/integration/test_text_type.py

```diff
@@ -0,0 +1,121 @@
+import re
+from playwright.sync_api import expect
+
+
+restrict_field_size_option_locator = "span:has-text('Restrict Field Size') input[type='checkbox']"
+field_size_limit_locator = "span:has-text('Field Size Limit') input[type='text']"
+
+
+def expect_table_to_open(page):
+    table_tab = page.locator("a[role=\"tab\"] >> text=All datatypes table")
+    expect(table_tab).to_be_visible()
+
+
+def get_column_header_locator(page, column_name):
+    column_header = f":nth-match(.table-content .header .cell .name:has-text('{column_name}'), 1)"
```

Comment:
`has-text` matches entities based on contains and not equality. In this scenario, we have two columns: 'char' and 'varchar', so trying to match 'char' will result in two headers being found.

While a good way is to have better names for columns, since this is a common method, it makes sense to use `nth-match of 1`, to always select just the first found element, similar to querySelector.

Reply:
> `has-text` matches entities based on contains and not equality

I think you could use `text-is` for an exact match, making the test a bit more strict.